### PR TITLE
Switch React import to * as React

### PR DESCRIPTION
**Pull request by xeoxx**
This change will allow the use of `"allowSyntheticDefaultImports": true` in tsconfig.json for dependent projects.
Without this change Interface evaluation uses empty (`{}`) React.HTMLElement interfaces for property evaluation. This causes e.g. `onClick` to not be recognized as a valid property on ListItem.

Checklist:

- [x] Test added / Snapshots updated
- [x] Documentation added / updated

### CODE REVIEW

**Comment by xeoxx** [comment]
> What about https://github.com/allthings/elements/blob/typescripted/tsconfig.json#L4? Can't we switch it to `false` too?

Unfortunately I was not able to resolve:
`import Mitt, { Handler } from 'mitt'`
`import json2mq from 'json2mq'`
`import matchSorter from 'match-sorter'`
None of them have a default export, so `import * as Mitt from 'mitt'` does not work.
If anyone has a workaround for that we could change `allowSyntheticDefaultImports=false`.

**Comment by yamafaktory** [comment]
@xeoxx What about just going for the ol'school:
```ts
 return `@media ${require('json2mq')(devices.map(device => MQ[device]))}`
```
https://github.com/allthings/elements/blob/5d9cf471209a90cd5cf7b199bc1c2629f85b909c/src/behaviour/Responsive.jsx#L17

**Comment by mosch** [comment]
I don't see why we shouldn't do this, however:
> This causes e.g. onClick to be not recognized as a valid property on ListItem.
Where do you see this happening?

**Comment by xeoxx** [comment]
> @xeoxx What about just going for the ol'school:
> ```ts
>  return `@media ${require('json2mq')(devices.map(device => MQ[device]))}`
> ```

**Comment by mosch** [comment]
Just realized that there are a bunch of comma changes – please make sure that they respect our linting rules.

File: doc/using-themes.md [hunk 1]
```diff
@@ -33,12 +34,13 @@ import { withTheme } from '@allthings/elements/behaviours/ThemeProvider'
...
 ```
-Simply wrap a component with *withTheme* and a property theme gets passed:
+
+Simply wrap a component with _withTheme_ and a property theme gets passed:
 
 ```js
 import { withTheme } from '@allthings/elements/behaviours/ThemeProvider'
 
-const Component = ({ theme }) => (
+const React.Component = ({ theme }) => (
```
Comment (dan-lee) [inline doc/using-themes.md]: Probably shouldn't be replaced

Comment (xeoxx) [reply doc/using-themes.md]: Reverted all `.md` changes

**Comment by xeoxx** [comment]
> @xeoxx What about just going for the ol'school:
> 
> ```ts
>  return `@media ${require('json2mq')(devices.map(device => MQ[device]))}`
> ```

I'd prefer not to do that, require makes it loose the type.

**Pull request merged**